Task.java

```diff
@@ -0,0 +1,69 @@
+public abstract class Task {
+    private String name;
+    private TaskStatus status;
+    private long schedule;
+    private Set<Task> dependsOn;
+
+    public abstract void runTask(EntityManager entityManager);
+
+    protected void finish(){
+        status = TaskStatus.Finished;
+        TaskManager.getInstance().taskFinished(this);
+    }
+
+    protected void reschedule(long timestamp){
+        schedule = timestamp;
+        TaskManager.getInstance().taskFinished(this);
```

Comment:
Why this method here and passing this?
If it is TaskManager all business logic should in the TaskManager class